### PR TITLE
Show column chart for categorical feature in data explorer

### PR DESCRIPTION
**Pull request by tongyu-microsoft**
This PR adds column chart when user selects a categorical feature with chart type aggregated plots in data explorer.

## Description
Before:
image

After:
image

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [x] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [ ] responsibleai/model-assessment

Python packages changed:

- [ ] erroranalysis
- [ ] raiutils
- [ ] raiwidgets
- [ ] rai_core_flask
- [ ] responsibleai

## Tests

- [ ] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by romanlutz** [comment]
@tongyu-microsoft the outlier point may be fixed if you pull in @xuke444 's latest merged PR (the one that sets the `key`).

I am a bit confused about the chart to be honest. The y-axis says relationship but it's values are numbers. Are they counts? If so, the axis label should say so I think. At the very least it should say "count of <feature>", i.e., "count of relationship."

Does that make sense or did I completely misinterpret this?

**Comment by tongyu-microsoft** [comment]
> @tongyu-microsoft the outlier point may be fixed if you pull in @xuke444 's latest merged PR (the one that sets the `key`).
> 
> I am a bit confused about the chart to be honest. The y-axis says relationship but it's values are numbers. Are they counts? If so, the axis label should say so I think. At the very least it should say "count of ", i.e., "count of relationship."
> 
> Does that make sense or did I completely misinterpret this?

Oh yes the outlier is fixed by Ke's change!! Thank you, saved me!

The y axis is count when user selects categorical features. Yes your question makes sense to me. We should probably update the y axis label, and maybe put "relationship" into legend.  Ilya also mentioned about it. This PR is the first step to at least show the plot. I will continue following up with Minsoo on how we could improve it. @minthigpen

**Review by imatiach-msft — APPROVED** [review]

File: libs/dataset-explorer/src/lib/getDatasetBar.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const xData = plotlyProps.data[0].x;
+  const xDataTypeCount =
+    chartProps?.xAxis.property &&
+    jointData.metaDict[chartProps.xAxis.property].sortedCategoricalValues
+      ?.length;
+
+  if (customData && xData) {
+    for (const [i, customDatum] of customData.entries()) {
+      if (!groupedData[(customDatum as any).Y]) {
+        groupedData[(customDatum as any).Y] = new Array(xDataTypeCount).fill(0);
```
Comment (imatiach-msft) [inline libs/dataset-explorer/src/lib/getDatasetBar.ts:31]: I see this several times:

```
(customDatum as any).Y
```
I think you can pull it out above as a variable inside the for loop at the top

File: libs/dataset-explorer/src/lib/getDatasetOption.ts [hunk 2]
```diff
@@ -20,5 +21,12 @@ export function getDatasetOption(
   if (chartProps?.chartType === ChartTypes.Scatter) {
     return getDatasetScatterOption(jointData, plotlyProps, chartProps);
   }
+  if (
+    chartProps?.yAxis.property &&
```
Comment (imatiach-msft) [inline libs/dataset-explorer/src/lib/getDatasetOption.ts:25]: nit: maybe move 
```
chartProps?.yAxis.property
```
to a separate variable to simplify:
```
const yAxisProp = chartProps?.yAxis.property
if (yAxisProperty && jointData.metaDict[yAxisProp].isCategorical || jointData.metaDict[yAxisProp].treatAsCategorical)
...
```

File: libs/dataset-explorer/src/lib/getDatasetBar.ts (hunk 1 above)
Comment (tongyu-microsoft) [reply libs/dataset-explorer/src/lib/getDatasetBar.ts]: Updated, thanks!

File: libs/dataset-explorer/src/lib/getDatasetOption.ts (hunk 2 above)
Comment (tongyu-microsoft) [reply libs/dataset-explorer/src/lib/getDatasetOption.ts]: Updated, thanks Ilya!

**Review by romanlutz — APPROVED** [review]

**Pull request merged**